public/app/features/search/components/DashboardSearch.tsx

```diff
@@ -22,17 +22,19 @@ export const DashboardSearch: FC<Props> = memo(({ onCloseSearch, folder }) => {
 
   // The main search input has own keydown handler, also TagFilter uses input, so
   // clicking Esc when tagFilter is active shouldn't close the whole search overlay
-  const onClose = () => {
-    onCloseSearch();
+  const onOverlayKeydown = (event: React.KeyboardEvent<HTMLElement>) => {
+    if (event.key === 'Escape' && (event.target as HTMLElement).tagName !== 'INPUT') {
+      onCloseSearch();
+    }
```

Comment:
I think we should handle this in keybindingSrv like all keyboard shortcuts (and other Esc key behaviors)

Reply:
But then we'd need to use `appEvents` to communicate with Search. Or do you mean to listen to events in `keybindingSrv` and update location query from there?